nix/shell.go

```diff
@@ -107,7 +110,7 @@ func rcfilePath(basename string) string {
 	return filepath.Join(home, basename)
 }
 
-func (s *Shell) Run(nixPath string) error {
+func (s *Shell) Run(nixShellFilePath string, srcDir string) error {
```

Comment:
I find this new interface for `Run` slightly confusing. Can you elaborate on why we now need both of these to be passed in?

Reply:
I use it to construct the directory-for-profile-binaries i.e. `srcDir + .devbox/profile/bin`

It seems that the intention is for `nix` package to simulate a go-sdk. In that case, this is maybe not the best API.